### PR TITLE
Make sure az_json_writer_append_json_text appends a comma between elements of an array.

### PR DESCRIPTION
**Pull request by ahsonkhan**
Fixes https://github.com/Azure/azure-sdk-for-c/issues/1574

### CODE REVIEW

**Comment by ahsonkhan** [comment]
cc @kevinior

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,10 @@
 # Release History
 
-## 1.1.0-beta.2 (Unreleased)
+## 1.1.0-beta.4 (Unreleased)
```
Comment (ahsonkhan) [inline CHANGELOG.md:3]: This was updated to be one higher than the version in the iot_pnp branch which is where earlier betas were shipped out of.
https://github.com/Azure/azure-sdk-for-c/blob/93fb3362a37753c67aca4b52b852441e097f4dbf/CHANGELOG.md#110-beta4-unreleased

FYI @danewalton, when you do your next sync.

**Comment by ahsonkhan** [comment]
@sima-zhu can you please help me understand where this broken link issue is coming from as part of the `Link verification check` and why it is getting flagged in this PR when it isn't modifying that markdown at all (and not in previous CI runs)? 
https://dev.azure.com/azure-sdk/public/_build/results?buildId=732609&view=logs&j=a129effc-2dd1-54d1-fb5a-ad7bdc0e851d&t=ccad100f-2904-5913-2083-c525ad929a03&l=40

```text
Summary of broken links:
'file:///D:/a/_work/1/s/sdk/samples/iot/docs/how_to_iot_hub_realtek_amebaD.md' has 1 broken link(s):
  ***/blob/master/sdk/samples/iot/aziot_realtek_amebaD
##[error]Found 222 links with 1 page(s) broken.
##[error]PowerShell exited with code '1'.
```

**Review by vhvb1989 — APPROVED** [review]
Does this mean we would want to release next March to publish this bug fix?

**Comment by ahsonkhan** [comment]
> Does this mean we would want to release next March to publish this bug fix?

I think so. But I will leave it up to Rick to decide, especially if we take it as a hot-fix for 1.0.1.

**Review by RickWinter — APPROVED** [review]

**Comment by RickWinter** [comment]
> @sima-zhu can you please help me understand where this broken link issue is coming from as part of the `Link verification check` and why it is getting flagged in this PR when it isn't modifying that markdown at all (and not in previous CI runs)?
> https://dev.azure.com/azure-sdk/public/_build/results?buildId=732609&view=logs&j=a129effc-2dd1-54d1-fb5a-ad7bdc0e851d&t=ccad100f-2904-5913-2083-c525ad929a03&l=40
> 
> ```
> Summary of broken links:
> 'file:///D:/a/_work/1/s/sdk/samples/iot/docs/how_to_iot_hub_realtek_amebaD.md' has 1 broken link(s):
>   ***/blob/master/sdk/samples/iot/aziot_realtek_amebaD
> ##[error]Found 222 links with 1 page(s) broken.
> ##[error]PowerShell exited with code '1'.
> ```

cc: @ewertons the link failure you had is showing up in master.  What was the fix you applied in your PR?
https://github.com/Azure/azure-sdk-for-c/pull/1568

**Comment by ahsonkhan** [comment]
@sima-zhu, @weshaggard - why is the generate release artifact passing here. Shouldn't the link verification step catch the same issue. I don't understand the fickleness
https://github.com/Azure/azure-sdk-for-c/pull/1602

https://dev.azure.com/azure-sdk/public/_build/results?buildId=733883&view=logs&j=a129effc-2dd1-54d1-fb5a-ad7bdc0e851d&t=ccad100f-2904-5913-2083-c525ad929a03
image

**Comment by ahsonkhan** [comment]
OK, I think it was because the parent of my branch was several months old. Hopefully the merge from master will fix the issue:
```text
commit 09617f2e7311f61ff32293a5e55aa55129361248
Author: Ahson Khan <ahson_ahmedk@yahoo.com>
Date:   Fri Oct 9 14:26:05 2020 -0700

    Add helper comment in clang format validation step to let contributor know                                                                                                                                                                                                 how to fix it (#1361)

```

**Comment by ahsonkhan** [comment]
Alright, syncing with master resolved the issue. My branch was just out-of-sync, and there is no link verification issue.

**Pull request merged**